cluster/cluster.py

```diff
@@ -276,12 +276,51 @@ def update(stack_name, version, force):
                            'UserDataWorker={}'.format(user_data_worker), 'KmsKey=*'])
     # wait for CF update to complete..
     subprocess.check_call(['senza', 'wait', '--timeout=600', stack_name, version])
-    perform_node_updates(stack_name, version, 'Master', user_data_master)
+    perform_node_updates(stack_name, version, 'Master', user_data_master, variables)
     wait_for_api_server(variables['api_server'])
-    perform_node_updates(stack_name, version, 'Worker', user_data_worker)
+    perform_node_updates(stack_name, version, 'Worker', user_data_worker, variables)
 
 
-def get_instances_in_service(group: dict):
+def get_k8s_nodes(api_server: str, token: str):
+    headers = { "Authorization": "Bearer {}".format(token) }
+    return requests.get(api_server + "/api/v1/nodes", headers=headers, timeout=5).json()
+
+
+def get_k8s_node_name(instance_id: str, config: dict):
+    nodes = get_k8s_nodes(config["api_server"], config["worker_shared_secret"])
+    for node in nodes["items"]:
+        if node["spec"]["externalID"] == instance_id:
+            return node["metadata"]["name"]
+    return ""
+
+
+def drain_node(node_name: str, config: dict, grace_period=10):
```

Comment:
in kubernetes each pod has a termination grace period after which the kubelet will try harder to kill it. this period is `30s` by default. i think it makes sense to match that default here.